lsp-semantic-tokens.el

```diff
@@ -250,12 +254,14 @@ ignoring the timer."
      `( :textDocument ,(lsp--text-document-identifier)
         ,@(if region (list :range (lsp--region-to-range (car region) (cdr region))) '()))
      (lambda (response)
-       (setq lsp--semantic-tokens-cache response)
-       (lsp-put lsp--semantic-tokens-cache :_documentVersion lsp--cur-version)
-       (lsp-put lsp--semantic-tokens-cache :_region region)
-       (when fontify-immediately (font-lock-flush))
-       ;; request full token set to improve fontification speed when scrolling
-       (when region (funcall request-full-token-set nil)))
+       (when (buffer-live-p semantic-tokenizing-buffer)
```

Comment:
did you run into a situation where this change made a difference? Because I think `:mode 'tick` should already evaluate the callback within the calling buffer, or cancel it altogether if the buffer is killed inbetween?

Reply:
I remember adding `buffer-live-p` because the following `with-current-buffer` failed indeed.

I don't remember the details right now, but I remember I _was_ pushing the limits for this code when I got that error. I personally say the cost is low (a little harder to read, more to think about) and the benefit is huge (user is not confounded by a hard-to-reproduce error), so this is worth in the long run.

Reply:
I think it's redundant, so I'd prefer removing it, thanks!

Reply:
done

Reply:
LGTM, thanks!